ext/http/00_serve.ts

```diff
@@ -313,27 +314,30 @@ class InnerRequest {
         path: this.#context.listener.addr.path,
       };
     }
-    if (this.#methodAndUri === undefined) {
+    if (this.#peerAddressAndPort === undefined) {
       if (this.#external === null) {
         throw new TypeError("Request closed");
       }
-      this.#methodAndUri = op_http_get_request_method_and_url(this.#external);
+      this.#peerAddressAndPort = op_http_get_request_peer_address_and_port(
+        this.#external,
+      );
     }
     return {
       transport: "tcp",
-      hostname: this.#methodAndUri[3],
-      port: this.#methodAndUri[4],
+      hostname: this.#peerAddressAndPort[0],
+      port: this.#peerAddressAndPort[1],
     };
   }
```

Comment:
I guess this would make accessing both `remoteAddr` and `url` slightly slower though? I think that's probably ok because this is accessed less often?